components/eventing-controller/pkg/backend/utils/eventmesh_utils.go

```
@@ -156,29 +142,80 @@ func ConvertKymaSubToEventMeshSub(subscription *eventingv1alpha2.Subscription, t
+		eventMeshSub.ContentMode = contentMode
+	}
+	// qos
+	if qosStr, ok := subscription.Spec.Config[eventingv1alpha2.ProtocolSettingsQos]; ok && qosStr != "" {
+		qos, err := getQos(qosStr)
+		if err != nil {
+			return err
+		}
+		eventMeshSub.Qos = qos
+	}
+	// handshake
+	if exemptHandshake, ok := subscription.Spec.Config[eventingv1alpha2.ProtocolSettingsExemptHandshake]; ok {
+		handshake, err := strconv.ParseBool(exemptHandshake)
+		if err != nil {
+			handshake = true
+		}
+		eventMeshSub.ExemptHandshake = handshake
+	}
+	return nil
+}
+
+// getEventMeshWebhookAuth uses default webhook auth unless specified in Subscription CR.
+func getEventMeshWebhookAuth(subscription *eventingv1alpha2.Subscription,
+	defaultWebhookAuth *types.WebhookAuth) (*types.WebhookAuth, error) {
+	auth := &types.WebhookAuth{}
+	// extract auth info from subscription CR if any
+	if authType, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthType]; ok {
+		if authType != string(types.AuthTypeClientCredentials) {
+			return nil, fmt.Errorf("invalid Type: %v, required: %v", authType,
+				string(types.AuthTypeClientCredentials))
+		}
+		auth.Type = types.AuthTypeClientCredentials
+	}
+
+	if tokenURL, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthTokenURL]; ok {
+		auth.TokenURL = tokenURL
+	}
+
+	if clientID, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthClientID]; ok {
+		auth.ClientID = clientID
+	}
+
+	if clientSecret, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthClientSecret]; ok {
+		auth.ClientSecret = clientSecret
+	}
+
+	if grantType, ok := subscription.Spec.Config[eventingv1alpha2.WebhookAuthGrantType]; ok {
+		if grantType != string(types.GrantTypeClientCredentials) {
+			return nil, fmt.Errorf("invalid GrantType: %v, required: %v", grantType,
+				string(types.GrantTypeClientCredentials))
+		}
+		auth.GrantType = types.GrantTypeClientCredentials
+	}
```

Does it make sense to move this check in the beginning of the function so it can fail fast?

---

updated